cmd/rpcdaemon/commands/zkevm_api.go

```diff
@@ -695,12 +694,11 @@ func (api *ZkEvmAPIImpl) GetBatchByNumber(ctx context.Context, batchNumber rpc.B
 	}
 	batch.BatchL2Data = batchL2Data
 
-	oldAccInputHash, err := api.l1Syncer.GetOldAccInputHash(ctx, &api.config.AddressRollup, api.config.L1RollupId, batchNo)
+	accInputHash, err := api.getAccInputHash(ctx, hermezDb, batchNo)
 	if err != nil {
-		log.Warn("Failed to get old acc input hash", "err", err)
-		batch.AccInputHash = common.Hash{}
+		return nil, fmt.Errorf("failed to get acc input hash for batch %d: %w", batchNo, err)
 	}
-	batch.AccInputHash = oldAccInputHash
+	batch.AccInputHash = *accInputHash
```

Comment:
worth a nil check here to avoid a potential panic?

Reply:
sure!